cmd/bffutil/main.go

```diff
@@ -193,6 +294,257 @@ func detailOrgs(c *cli.Context) (err error) {
+	appdata = &auth.AppMetadata{}
+	if err = appdata.Load(user.AppMetadata); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	if permissions, err = auth0.User.Permissions(*user.ID); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	if !HasPermission(auth.SwitchOrganizations, permissions) {
+		return cli.Exit("the user must be a TSP user", 1)
+	}
+
+	// Ask if we should proceed
+	username, _ = auth.UserDisplayName(user)
+	if !askForConfirmation(fmt.Sprintf("create org for TestNet: %s and MainNet: %s records with user %s?", testname, mainname, username)) {
+		return cli.Exit("canceled at request of user", 0)
+	}
+
+	// Create new organization record
+	org := &models.Organization{
+		Name:      c.String("name"),
+		Domain:    c.String("domain"),
+		CreatedBy: "support@rotational.io",
+	}
+
+	// TODO: should we make domain required?
+	if org.Domain != "" {
+		if org.Domain, err = bff.NormalizeDomain(org.Domain); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		if err = bff.ValidateDomain(org.Domain); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		// TODO: Check for duplicate domains
+	}
+
+	// Add the user as a collaborator to the organization
+	// NOTE: expecting the user to be a TSP so no roles are modified
+	collaborator := &models.Collaborator{
+		Email:    *user.Email,
+		UserId:   *user.ID,
+		Verified: *user.EmailVerified,
+	}
+
+	if err = org.AddCollaborator(collaborator); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Add the directory records
+	if mainnetVASP != nil {
+		org.Mainnet = &models.DirectoryRecord{
+			Id:                  mainnetVASP.Id,
+			RegisteredDirectory: mainnetVASP.RegisteredDirectory,
+			CommonName:          mainnetVASP.CommonName,
+			Submitted:           mainnetVASP.FirstListed,
+		}
+	}
+
+	if testnetVASP != nil {
+		org.Testnet = &models.DirectoryRecord{
+			Id:                  testnetVASP.Id,
+			RegisteredDirectory: testnetVASP.RegisteredDirectory,
+			CommonName:          testnetVASP.CommonName,
+			Submitted:           testnetVASP.FirstListed,
+		}
+	}
+
+	// Create the registration form
+	var vasp *pb.VASP
+	if mainnetVASP != nil {
+		vasp = mainnetVASP
+	} else if testnetVASP != nil {
+		vasp = testnetVASP
+	}
+
+	if vasp != nil {
+		reg := models.NewRegisterForm()
+		reg.Website = vasp.Website
+		reg.BusinessCategory = vasp.BusinessCategory
+		reg.VaspCategories = vasp.VaspCategories
+		reg.EstablishedOn = vasp.EstablishedOn
+		reg.OrganizationName = org.Name
+		reg.Entity = vasp.Entity
+		reg.Contacts = vasp.Contacts
+		reg.Trixo = vasp.Trixo
+
+		if mainnetVASP != nil {
+			reg.Mainnet = &models.NetworkDetails{
+				CommonName: mainnetVASP.CommonName,
+				Endpoint:   mainnetVASP.TrisaEndpoint,
+			}
+		}
+
+		if testnetVASP != nil {
+			reg.Testnet = &models.NetworkDetails{
+				CommonName: testnetVASP.CommonName,
+				Endpoint:   testnetVASP.TrisaEndpoint,
+			}
+		}
+
+		// TODO: help, what should the form steps be?
+		reg.State.Current = 1
+		reg.State.ReadyToSubmit = reg.ReadyToSubmit("all")
```

Comment:
My suggestion is to mark the steps as in progress so that the user still has to click through the steps and confirm the information, however we may need to test this to make sure this doesn't break anything on the frontend form.

```json
"steps":[
    {"key":1,"status":"progress"},
    {"key":2,"status":"progress"},
    {"key":3,"status":"progress"},
    {"key":4,"status":"progress"},
    {"key":5,"status":"progress"},
    {"key":6,"status":"progress"}
]
```

I think we also need to set the `started` timestamp in the state to the current time.

Reply:
I see what you mean, but the registration has already been submitted most likely. Should I mark it as done instead?